NCGO2024/Assets/App/Resource/Scripts/UI/NetworkedPlayerData.cs

```diff
@@ -0,0 +1,131 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Unity.Netcode;
+using System;
+public class NetworkedPlayerData : NetworkBehaviour
+{
+    public NetworkList<PlayerInfoData> _allConnectedPlayers;
+    private int _players = -1;
+    private ulong _serverLocalID;
+
+    private Color[] _PlayerColors = new Color[]
+    {
+        Color.blue, Color.magenta, Color.red, Color.green, Color.yellow
+    };
+
+
+
+    private void Awake()
+    {
+        _allConnectedPlayers = new NetworkList<PlayerInfoData>(readPerm: NetworkVariableReadPermission.Everyone);
+    }
+
+
+
+    public override void OnNetworkSpawn()
+    {
+        base.OnNetworkSpawn();
+
+        if (!IsServer) return;
+
+        NetworkManager.Singleton.OnConnectionEvent += OnConnectionEvents;
+        _serverLocalID = NetworkManager.ServerClientId;
+
+    }
+
+    public override void OnNetworkDespawn()
+    {
+
+        if (IsServer) 
+        { 
+        
+        
+            NetworkManager.Singleton.OnConnectionEvent -= OnConnectionEvents;
+        }
+
+        base.OnNetworkDespawn();
+    }
+
+
+
+    private void OnConnectionEvents(NetworkManager netManager, ConnectionEventData eventData)
+    {
+        if (eventData.EventType == ConnectionEvent.ClientConnected)
+        {
+
+            CreateNewClientData(eventData.ClientId);
+        }
+
+        if (eventData.EventType == ConnectionEvent.ClientDisconnected)
+        {
+            //RemovePlayerData();
```

Comment:
Need to finish this implementation or uncomment remove function at kick in the LobbyManager